core/src/com/unciv/logic/civilization/CivilizationInfo.kt

```diff
@@ -160,6 +168,13 @@ class CivilizationInfo {
         return newResourceSupplyList
     }
 
+    fun getViewableResources(): Sequence<TileResource> =
+            gameInfo.ruleSet.tileResources.values
+                    .filter { it.revealedBy == null || tech.isResearched(it.revealedBy!!) }
+                    .asSequence()
+
+    fun isCapitalConnectedToCity(city: CityInfo): Boolean = citiesConnectedToCapitalToMediums.keys.contains(city)
```

Comment:
I'm pretty sure we have a function like this somewhere, probably in the cityInfo itself - isConnectedToCapital